app/views/users/show.html.erb

```diff
@@ -1,5 +1,21 @@
 <section class="user-section">
-  <h2><%= "Name: #{@user.name}" %></h2>
+  <div class="user-header">
+    <h2><%= "Name: #{@user.name}" %></h2>
+
+    <% if current_user.id == @user.id %>
+    <% elsif friend? @user %>
+      <h3> is your friend </h3>
+    <% elsif request_sent? @user %>
+      <h3> Friend request sent </h3>
+    <% elsif request_received_from?(@user) && request_received_from?(@user).status == 'Pending' %>
+      <h3> Is waiting for your response </h3>
+    <% elsif request_received_from?(@user) && request_received_from?(@user).status == 'Rejected' %>
+      <h3>Friend request rejected</h3>
+    <% else %>
+      <%= button_to 'Send friend request', { :controller => "users", :action => "send_friend_request", :user_id => @user.id , :method=>:post }  %>
+    <% end %>
```

Comment:
- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)